src/main/java/data/plans/PlanList.java

```diff
@@ -117,40 +117,27 @@ public int getIndexNumFromPlanName(String planName) throws InvalidPlanException
     public Plan createAndAddPlan(String userArgument) throws ArrayIndexOutOfBoundsException,
             NumberFormatException, InvalidPlanException {
         String userPlanNameInput = userArgument.split(PlanCommand.CREATE_ACTION_WORKOUTS_KEYWORD)[0].trim();
-
-        String className = this.getClass().getSimpleName();
-        boolean hasValidPlanName = checkPlanNameValidity(userPlanNameInput);
-        if (!hasValidPlanName) {
-            logger.log(Level.WARNING, "Plan name is invalid.");
-            throw new InvalidPlanException(className, InvalidPlanException.INVALID_PLAN_NAME_ERROR_MSG);
-        }
-
         String userWorkoutNumbersString = userArgument.split(PlanCommand.CREATE_ACTION_WORKOUTS_KEYWORD)[1].trim();
+        String className = this.getClass().getSimpleName();
+        checkPlanNameValidity(userPlanNameInput, className);
 
         int numberOfWorkoutsInAPlan = userWorkoutNumbersString.split(",").length;
-        boolean isAppropriateNumberOfWorkouts = checkMinMaxNumberOfWorkouts(numberOfWorkoutsInAPlan);
-        if (!isAppropriateNumberOfWorkouts) {
-            logger.log(Level.WARNING, "Number of workouts to add in a plan is invalid.");
-            throw new InvalidPlanException(className, InvalidPlanException.MIN_MAX_WORKOUTS_IN_A_PLAN);
-        }
+        checkMinMaxNumberOfWorkouts(numberOfWorkoutsInAPlan, className);
         assert (numberOfWorkoutsInAPlan > 0 && numberOfWorkoutsInAPlan <= MAX_NUMBER_OF_WORKOUTS_IN_A_PLAN);
```

Comment:
Perhaps each condition can be enclosed with its own parentheses to make the two conditions more distinguishable from each other?